lib/rsa/accumulator.rb

```diff
@@ -59,10 +59,7 @@ def add(*elements)
       p = elements_to_prime(elements)
       self.value = value.pow(p, n)
       if hold_elements
-        elements.each do |e|
-          p = hash_to_prime(e)
-          self.products *= p
-        end
+        self.products *= p
```

Comment:
good catch. You can write more like Ruby.

`self.products *= p if hold_elements`